hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/RolloutListGrid.java

```diff
@@ -162,40 +183,55 @@ void onEvent(final RolloutEvent event) {
     }
 
     /**
+     * Handles the RolloutDeletedEvent to refresh the grid.
+     *
+     * @param eventContainer
+     *            container which holds the rollout delete event
+     */
+    @EventBusListenerMethod(scope = EventScope.UI)
+    public void onRolloutDeletedEvent(final RolloutDeletedEventContainer eventContainer) {
+        refreshContainer();
+    }
+
+    /**
      * Handles the RolloutChangeEvent to refresh the item in the grid.
      *
      * @param eventContainer
      *            container which holds the rollout change event
      */
-    @SuppressWarnings("unchecked")
     @EventBusListenerMethod(scope = EventScope.UI)
     public void onRolloutChangeEvent(final RolloutChangeEventContainer eventContainer) {
         eventContainer.getEvents().forEach(this::handleEvent);
     }
 
     private void handleEvent(final RolloutChangeEvent rolloutChangeEvent) {
-        if (!rolloutUIState.isShowRollOuts()) {
+        if (!rolloutUIState.isShowRollOuts() || rolloutChangeEvent.getRolloutId() == null) {
             return;
         }
-        final Rollout rollout = rolloutManagement.findRolloutWithDetailedStatus(rolloutChangeEvent.getRolloutId());
-        final TotalTargetCountStatus totalTargetCountStatus = rollout.getTotalTargetCountStatus();
-        final LazyQueryContainer rolloutContainer = (LazyQueryContainer) getContainerDataSource();
-        final Item item = rolloutContainer.getItem(rolloutChangeEvent.getRolloutId());
-        if (item == null) {
-            refreshContainer();
-            return;
+        final Rollout rollout = rolloutManagement.findRolloutWithDetailedStatus(rolloutChangeEvent.getRolloutId(),
+                false);
+
+        // rollout is null if rollout was deleted
+        if (rollout != null) {
```

Comment:
Would prefer a early return and remove the comment and a meaningful private methode